contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {IPCVGuardian} from "./IPCVGuardian.sol";
+import {IPCVDeposit} from "./IPCVDeposit.sol";
+import {CoreRefPauseableLib} from "../libs/CoreRefPauseableLib.sol";
+import {TribeRoles} from "../core/TribeRoles.sol";
+
+/// @notice PCV Guardian is a contract to safeguard protocol funds
+/// by being able to withdraw whitelisted PCV deposits to an immutable safe address
+contract PCVGuardian is IPCVGuardian, CoreRef {
+    using CoreRefPauseableLib for address;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private whitelistAddresses;
+
+    ///@notice safe address where PCV funds can be withdrawn to
+    address public immutable safeAddress;
+
+    constructor(
+        address _core,
+        address _safeAddress,
+        address[] memory _whitelistAddresses
+    ) CoreRef(_core) {
+        safeAddress = _safeAddress;
+
+        // improbable to ever overflow
+        unchecked {
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _addWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    modifier onlyWhitelist(address pcvDeposit) {
+        require(
+            isWhitelistAddress(pcvDeposit),
+            "Provided address is not whitelisted"
```

Comment:
Error messages should have the contract name at the start. Message should be `PCVGuardian: Provided address is not whitelisted`